### ISSUE TITLE
Adding (and removing?) applications does not work, throws an error

### ISSUE THREAD

**Issue by Faalagorn** [jlHertel/adriconf#13]
As of the most recent (`c177182`) commit, I can't add a new application anymore. I initially thought that adding the shortcuts was what broke it, but after downgrading to `3ac524b` the issue is still there, so it wasn't related to #12 . Trying to add it with either clicking the button or hitting `Ctrl+D` shortcut results in the following error. I didn't try to remove any of the built-in applications.

```
(adriconf:5104): gtkmm-CRITICAL **: 16:56:29.912: gtkmm: object `newAppDialog' not found in GtkBuilder file.

(adriconf:5104): gtkmm-CRITICAL **: 16:56:29.912: gtkmm: Gtk::Builder: widget `newAppDialog' was not found in the GtkBuilder file, or the specified part of it.

** (adriconf:5104): CRITICAL **: 16:56:29.912: Gtk::Builder::get_widget(): dynamic_cast<> failed.
Add Application dialog is not in glade file!
```

* Arch Linux (64-bit)
* Linux 4.16-12
* mesa-git 79fe00efb4
* AMD Radeon RX 580

**Comment by jlHertel** [comment]
This error indicates a missing window in the glade file.
Maybe you edited it?
My hint here would be to get the latest code at git, deleting your resources.c a rebuild the software.
If it still doesn't works, please give me more information about the way you are getting the software.

**Comment by Faalagorn** [comment]
Ah, thanks for the reply! I'm no coder, with relatively little Linux experience yet, so it was hard for me to troubleshoot, that's the package I've been using: https://aur.archlinux.org/packages/adriconf-git/ – if that's a packaging issue, guess I can drop a comment there for a maintainer to fix it.

**Comment by jlHertel** [comment]
I have checked the build, and doing one build over another really break things if the glade file has changed. On the positive side, running `make clean` removes the old files, and so this can be a possible solution.
I'm not exactly sure, but I suspect the problem is that at the end of the build, the build directory `adriconf-build` is left on the filesystem, so updates will work on these old files.

@ChristophHaag Is possible to fix your PKGBUILD so that it removes the build directory at the end of the build?
If not, an alternative solution is to just run a `make clean` on the build dir.

**Comment by ChristophHaag** [comment]
Hmm I thought when you do
```
add_custom_command(OUTPUT ${CMAKE_SOURCE_DIR}/resources.c
.....
}
set_source_files_properties(${CMAKE_SOURCE_DIR}/resources.c PROPERTIES
    GENERATED TRUE
)
```
then cmake is supposed to rerun the command when command generating resources.c changes...

**Comment by jlHertel** [comment]
@ChristophHaag you are right, cmake should build this.
I checked it again and this command was missing the `DEPENDS` part. I fixed this with a937a80.
